utils/users.py

```diff
@@ -79,6 +80,45 @@ def create_refresh_token(subject: Union[str, Any], expires_delta: int = None) ->
     return encoded_jwt, expires_delta
 
 
+async def get_refresh_user(token: str = Depends(oauth2_scheme)):
+    """ Get user by long term token """
+
+    try:
+        payload = jwt.decode(
+            token, JWT_REFRESH_SECRET_KEY, algorithms=[ALGORITHM]
+        )
+        token_data = users_schema.TokenPayload(**payload)
+
+        if datetime.datetime.fromtimestamp(token_data.exp) < datetime.datetime.now():
+            raise HTTPException(
+                status_code=status.HTTP_401_UNAUTHORIZED,
+                detail="Token expired",
+                headers={"WWW-Authenticate": "Bearer"},
+            )
+    except(jwt.JWTError, ValidationError):
+        raise HTTPException(
+            status_code=status.HTTP_403_FORBIDDEN,
+            detail="Could not validate credentials",
+            headers={"WWW-Authenticate": "Bearer"},
+        )
+    user = await users_crud.get_user_by_email(payload['sub'])
```

Comment:
you don't really need to query user here, you can just use email from payload to generate access token

Reply:
Done